### PR TITLE
feat(kit): `CheckboxBlock` support center alignment with `align-items` CSS

### LINKED ISSUE #4259: 🐞 - Checkbox Block is unable to adapt to a custom height size
### Playground Link

https://stackblitz.com/edit/angular-m8v7na?file=src%2Fapp%2Fapp.component.html,src%2Fapp%2Fapp.component.less&hideExplorer=1

### Description

The [CheckboxBlock](https://taiga-ui.dev/components/checkbox-block) component has non-adapted checkbox positioning to a custom height size.
https://github.com/Tinkoff/taiga-ui/blob/main/projects/kit/components/checkbox-block/checkbox-block.style.less#L86-L116

It causes the following issue

checkbox-block-l

### Angular version

12.2.7

### Taiga UI version

3.25.0

### Which browsers have you used?

- [X] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### Which operating systems have you used?

- [ ] macOS
- [X] Windows
- [ ] Linux
- [ ] iOS
- [ ] Android

### PR DESCRIPTION
**Pull request by waterplea**
## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Refactoring
- [ ] Build or CI related changes
- [ ] Documentation content changes

## What is the current behavior?

Closes #4259

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

**Review by splincode — APPROVED** [review]

File: projects/kit/components/checkbox-block/checkbox-block.style.less
[the remark this answers is not in the document]
Comment (waterplea) [reply projects/kit/components/checkbox-block/checkbox-block.style.less:1]: True. But we do not have anything like that yet and I don't want to introduce it just yet.

**Comment by constantant** [comment]
> Visit the preview URL for this PR (updated for commit [fc090a8](https://github.com/Tinkoff/taiga-ui/commit/fc090a8287849a94664b8faab042bdd16dc0a3b2)):
> 
> https://taiga-ui--pr4370-checkbox-ozrwdeif.web.app
> 
> (expires Thu, 04 May 2023 13:27:21 GMT)
> 
> 🔥 via [Firebase Hosting GitHub Action](https://github.com/marketplace/actions/deploy-to-firebase-hosting) 🌎
> 
> Sign: 4b5ece1e114386f6a105425ef799091475b249eb

It does not work in case you make height smaller :(
image

**Review by constantant — CHANGES_REQUESTED** [review]
The feature does not provide expected behaviour. Look at the [previous comment](https://github.com/Tinkoff/taiga-ui/pull/4370#issuecomment-1533041593)

File: projects/kit/components/multi-select/multi-select.component.ts
Comment (constantant) [inline projects/kit/components/multi-select/multi-select.component.ts:1]: Is that change relevant?

**Review by vladimirpotekhin — APPROVED** [review]

**Pull request merged**